src/Foundatio.RabbitMQ/Messaging/RabbitMQMessageBus.cs

```diff
@@ -117,21 +131,22 @@ public class RabbitMQMessageBus : MessageBusBase, IMessageBus {
                 return;
             _logger.Trace("Message Publish: {messageType}", messageType.FullName);
 
-            // RabbitMQ only supports delayed messages with a third party plugin.
-            if (delay.HasValue && delay.Value > TimeSpan.Zero) {
-                await AddDelayedMessageAsync(messageType, message, delay.Value).AnyContext();
-                return;
-            }
-
             var data = await _serializer.SerializeAsync(new MessageBusData {
                 Type = messageType.AssemblyQualifiedName,
                 Data = await _serializer.SerializeToStringAsync(message).AnyContext()
             }).AnyContext();
             
             var basicProperties = _publisherChannel.CreateBasicProperties();
             basicProperties.Persistent = _persistent;
-            basicProperties.Expiration = _defaultMessageTimeToLive.Milliseconds.ToString();
-
+            basicProperties.Expiration = _defaultMessageTimeToLive.TotalMilliseconds.ToString(CultureInfo.InvariantCulture);
+            // RabbitMQ only supports delayed messages with a third party plugin called "rabbitmq_delayed_message_exchange"
+            if (_delayedExchange && delay.HasValue && delay.Value > TimeSpan.Zero) {
```

Comment:
Looked at that link you post, any disadvantages to having two exchanges defined with the same exact name? If we fail to create or publish to the one we dispose of it and set it to null and call one for no delay, and the other or base implementation for delayed messages?

By doing one or the other right now and taking a perf hit kinda sucks, especially when it might be total random when you want to do a delayed message...